### PR TITLE
56 canvas painter

### LINKED ISSUE #56: Refactor to canvasPainter

### PR DESCRIPTION
**Pull request by stevepentler**
closes #56 

@jecrockett - refactor partner
@rrgayhart  - instructor
@dastinnette - member of another refactor team, please review!

What's this PR do?
Abstracts draw object functionality into a canvasPainter object. This removes responsibilities from the generateFish function in the game class.

Where should the reviewer start?
The canvasPainter object is exported Game object, then called in the drawObjects function rather than the generateFish function. This is more appropriate for single responsibility principles.  

How should this be manually tested?
This method draws to the canvas, which is hard to test. However, extracting this functionality will improve our ability to test generateFish and other fish properties that are not canvas related. 

Any background context you want to provide?
This is step one of the canvasPainter object. Eventually, the canvasPainter will be responsible for drawing the boat and bullet objects as well. 

Screenshots (if appropriate)

What gif best describes this PR or how it makes you feel?
http://cdn4.gurl.com/wp-content/uploads/2013/08/tumblr_inline_mqkivpZMyc1qz4rgp.gif

### CODE REVIEW

File: lib/game.js [hunk 1]
```diff
@@ -34,10 +36,14 @@ function drawOcean() {
   context.drawImage(images.ocean, 0, 135);
 };
 
+function drawObjects() {
+  canvasPainter.drawFish(fishies);
+}
+
 function generateFish() {
+  drawObjects();
```
Comment (jecrockett) [inline lib/game.js]: @stevepentler This got it working, but drawObjects shouldn't live in generateFish full time. Pull out to the game loop once the canvas painter can draw everything.

**Comment by rrgayhart** [comment]
Don't forget the inline image syntax. I need to see Bubba immediately. (`` instead of `[]()`)

**Comment by rrgayhart** [comment]
:+1: Nice refactor!

**Comment by stevepentler** [comment]
@rrgayhart Thank you for the gif feedback. Another crucial issue closed!

**Pull request merged**